robusta_krr/core/runner.py

```diff
@@ -213,11 +216,14 @@ async def _check_data_availability(self, cluster: Optional[str]) -> None:
                 }
             )
 
-    async def _gather_object_allocations(self, k8s_object: K8sObjectData) -> ResourceScan:
+    async def _gather_object_allocations(self, k8s_object: K8sObjectData) -> Optional[ResourceScan]:
         recommendation = await self._calculate_object_recommendations(k8s_object)
 
         self.__progressbar.progress()
 
+        if recommendation is None:
+            return None
```

Comment:
does this remove the row from the scan?

Reply:
Yes, but that is only the case if something failed (currently with prometheus).

It would have been better to use exceptions here, but as we are using 3.9 and we don't yet have ExceptionGroups from later versions, it is easier to return a value

In case this fails for part of rows, we will also have error logs (for example one cluster does not work, another does: we will show only sucessfull results, skipping the failed cluster with an error log)